### PR TITLE
KPV-2305 :: Refresh contest application list when sorted by votes.

### PR DESCRIPTION
**Pull request by pverdu-kuorum**

### CODE REVIEW

File: web-app/js/campaigns/contest.js [hunk 1]
```diff
@@ -21,10 +21,22 @@ $(function () {
...
-        var listSelector = $a.attr("data-listSelector");
-        contestApplicationHelper.showListApplications(listSelector);
+        if(!$a.parent().hasClass("active")) {
+            $a.parents("ul.nav").find("li").removeClass("active")
+            $a.parent().addClass("active")
+            var listSelector = $a.attr("data-listSelector");
+
+            //If sorted by number of votes is selected, data is refreshed.
+            if(listSelector === "vote") {
+                var $ul = $("#" + contestApplicationHelper.containerId).find(".search-list." + listSelector);
```
Comment (DavidSanFer555) [inline web-app/js/campaigns/contest.js]: Put this block of actions on a method. Something like clearList(selector)

File: web-app/js/campaigns/contest.js [hunk 2]
```diff
@@ -21,10 +21,22 @@ $(function () {
...
-        $a.parent().addClass("active")
-        var listSelector = $a.attr("data-listSelector");
-        contestApplicationHelper.showListApplications(listSelector);
+        if(!$a.parent().hasClass("active")) {
+            $a.parents("ul.nav").find("li").removeClass("active")
+            $a.parent().addClass("active")
+            var listSelector = $a.attr("data-listSelector");
+
+            //If sorted by number of votes is selected, data is refreshed.
+            if(listSelector === "vote") {
```
Comment (DavidSanFer555) [inline web-app/js/campaigns/contest.js]: Remove this condition, It isn't needed

File: web-app/js/campaigns/contest.js [hunk 3]
```diff
@@ -21,10 +21,22 @@ $(function () {
         e.preventDefault();
         var $a = $(this);
         console.log($a)
-        $a.parents("ul.nav").find("li").removeClass("active")
-        $a.parent().addClass("active")
-        var listSelector = $a.attr("data-listSelector");
```
Comment (DavidSanFer555) [inline web-app/js/campaigns/contest.js:26]: Try to use const instead var when posible. It reduces scope and reassignment problems

**Review by DavidSanFer555 — CHANGES_REQUESTED** [review]

File: web-app/js/campaigns/contest.js (hunk 1 above)
Comment (pverdu-kuorum) [reply web-app/js/campaigns/contest.js]: Putted into new function called setActiveAndRefreshList.

File: web-app/js/campaigns/contest.js (hunk 2 above)
Comment (pverdu-kuorum) [reply web-app/js/campaigns/contest.js]: Removed

File: web-app/js/campaigns/contest.js (hunk 3 above)
Comment (pverdu-kuorum) [reply web-app/js/campaigns/contest.js:26]: Changed the var to const type variables.

**Review by pverdu-kuorum — COMMENTED** [review]
Refactor applied.

**Review by DavidSanFer555 — APPROVED** [review]

**Pull request merged**